### PR TITLE
Remove json() utility

### PR DESCRIPTION
**Pull request by brophdawg11**
This was deprecated with Single Fetch in Remix v2, and folks can always use `Response.json()` if needed.

* First commit contains functional changes for easier review
* Second/Third contain updates to unit and integration tests

**TODO:**
* [x] Figure out `types.ts` issue mentioned in https://github.com/remix-run/react-router/pull/12146/files#r1805311358

### CODE REVIEW

File: packages/react-router/lib/dom/ssr/data.ts [hunk 1]
```diff
@@ -5,16 +5,6 @@ import "../global";
...
 
-export function isResponse(value: any): value is Response {
-  return (
-    value != null &&
-    typeof value.status === "number" &&
-    typeof value.statusText === "string" &&
-    typeof value.headers === "object" &&
-    typeof value.body !== "undefined"
-  );
-}
```
Comment (brophdawg11) [inline packages/react-router/lib/dom/ssr/data.ts:16]: We had this duped in `@remix-run/router` and `remix-run/server-runtime` so we can share the router one now

File: packages/react-router/lib/router/utils.ts [hunk 2]
```diff
@@ -1306,26 +1305,6 @@ export const normalizeSearch = (search: string): string =>
...
-  let headers = new Headers(responseInit.headers);
-  if (!headers.has("Content-Type")) {
-    headers.set("Content-Type", "application/json; charset=utf-8");
-  }
-
-  return new Response(JSON.stringify(data), {
-    ...responseInit,
-    headers,
-  });
-};
```
Comment (brophdawg11) [inline packages/react-router/lib/router/utils.ts:1327]: 👋

File: packages/react-router/lib/server-runtime/responses.ts [hunk 3]
```diff
@@ -1,81 +0,0 @@
-import {
```
Comment (brophdawg11) [inline packages/react-router/lib/server-runtime/responses.ts:1]: Everything in here was duped in the router so we use those now

File: packages/react-router/lib/types.ts [hunk 4]
```diff
@@ -170,11 +167,12 @@ type __tests = [
     >
   >,
   Expect<
+    //@ts-expect-error TODO: Why is this failing?
```
Comment (brophdawg11) [inline packages/react-router/lib/types.ts:170]: I'm unsure what's going on here.  We no longer have any sort of `TypedResponse` so I would have expected a raw json value to make this pass?

File: packages/react-router/lib/types.ts [hunk 5]
```diff
@@ -191,11 +189,12 @@ type __tests = [
...
```
Comment (brophdawg11) [inline packages/react-router/lib/types.ts:192]: Same thing here

File: packages/react-router/lib/types.ts (hunk 4 above)
Comment (brookslybrand) [reply packages/react-router/lib/types.ts:170]: I think this is still expecting the `Date` to get serialized. Looks like it should be:

```ts
  Expect<
    Equal<
      Pretty<
        ServerDataFrom<
          () =>
            | { json: string; b: Date; c: () => boolean }
            | DataWithResponseInit<{ data: string; b: Date; c: () => boolean }>
        >
      >,
      | { json: string; b: Date; c: undefined }
      | { data: string; b: Date; c: undefined }
    >
  >,
```

I'm having trouble getting GitHub to do the suggested commit thing

File: packages/react-router/lib/types.ts (hunk 5 above)
Comment (brookslybrand) [reply packages/react-router/lib/types.ts:192]: Again, struggling to get GitHub to add a commit suggestion, but should be this

```ts
Expect<
    Equal<
      Pretty<
        ClientDataFrom<
          () =>
            | { json: string; b: Date; c: () => boolean }
            | DataWithResponseInit<{ data: string; b: Date; c: () => boolean }>
        >
      >,
      | { json: string; b: Date; c: () => boolean }
      | { data: string; b: Date; c: () => boolean }
    >
  >,
```

File: packages/react-router/lib/types.ts (hunk 4 above)
Comment (brophdawg11) [reply packages/react-router/lib/types.ts:170]: ohhhhh thank you I totally missed the Date in there

**Pull request merged**